CODE_STYLE.md

```diff
@@ -126,6 +126,25 @@ Use `re_error::format(err)` when displaying an error.
 
 
 ## C++
+For compatibility reasons we use C++11 (and nothing newer).
```

Comment:
This is sadly very wrong. We use C++17 all over the place. We'll need to make a lot of changes to get to C++11 compatibility at this point

Reply:
the most notable thing here isn't even stuff like std::optional, it is that we rely on the enhanced type deduction a lot

Reply:
oh no :(

Reply:
it's kinda hard to avoid for us with arrow forcing c++17 compilation upon us